### DISCUSSION TITLE
Attestation format

### DISCUSSION THREAD

**Discussion by sb230132** [Yubico/libfido2#594] [Q&A] [answered]
Hi

During registration what value should i set in fido_cred_fmt() ?
If i test with yubikey 5 series device which has FIDO2 protocol enabled i need to set to "packed".
For yubikey 4 series device which has only U2F protocol iam setting "fido-u2f".

I cannot hard code a value in my RP server, because user can have any kind of usb device ?

Regards
Satish.B

**Comment by martelletto** [comment, marked as answer]
Hi @sb230132,

That depends on the protocol spoken between the FIDO2 client and the RP.

In WebAuthn, the attestation statement format identifier, the attestation statement, and authdata (returned by [fido_cred_fmt()](https://developers.yubico.com/libfido2/Manuals/fido_cred_fmt.html), [fido_cred_attstmt_ptr()](https://developers.yubico.com/libfido2/Manuals/fido_cred_attstmt_ptr.html), and [fido_cred_authdata_ptr()](https://developers.yubico.com/libfido2/Manuals/fido_cred_authdata_ptr.html) respectively) are passed by the client to the RP during registration/enrollment as part of the [attestation object]( https://www.w3.org/TR/webauthn-2/#attestation-object). Sample code to build an attestation object using libfido2 can be found at https://github.com/martelletto/fido2-webauthn-client/blob/master/cbor.c#L175-L241.

Kind regards,

-p.

**Reply by sb230132** [reply]
Ok, So what i understand is that the format which is present in attestation object is passed from client to RP server. Thus its not correct to hard code a value in RP server. 
Am i correct ?

**Reply by martelletto** [reply]
Yes, that's correct.

**Reply by sb230132** [reply]
Thank you.
